### ISSUE TITLE
error: invalid conversion from 'void*' to 'double*'

### ISSUE THREAD

**Issue by smfarzaneh** [FFTW/fftw3#197]
There is a compile error during installation after the `make` command. 

```
fftw-bench.c: In function 'void setup(bench_problem*)':
fftw-bench.c:215:43: error: invalid conversion from 'void*' to 'double*' [-fpermissive]
  215 |           BENCH_ASSERT(FFTW(alignment_of)(ptr) == 0);
      |                                           ^~~
      |                                           |
      |                                           void*
```

I was able to resolve this in `fftw-bench.c` file by replacing 
```
void *ptr = FFTW(malloc(42));
```
with
```
double *ptr = (double *)FFTW(malloc(42));
```

**Comment by matteo-frigo** [comment]
Implicit conversion void* -> T* is perfectly valid C.  What compiler/flags are you using?  How did you invoke "configure"?

**Comment by smfarzaneh** [comment]
Sorry for the incomplete info. I only used the `CC` flag to set my compiler to GCC 9.3.0 as follows `./configure CC=g++-9`. 
The system is a MacOS Catalina with latest gcc version 9.3.0 installed by using Homebrew.

**Comment by matteo-frigo** [comment]
Try './configure CC=gcc-9'

**Comment by matteo-frigo** [comment]
Fixed in 3a103f42

Note that FFTW is a C program, not a C++ program.  Historically we have made an effort to make it compile with both C and C++, precisely to avoid situations like yours, but we don't really try too hard.  Even though after the fix FFTW should compile as a C++ program, you aren't really supposed to do it (you get a different binary that we have never tested---C++ and C differ in many subtle ways).

In the old days, g++ foo.c would compile foo.c as a C program.  It's good to know that this is no longer true, and that g++ foo.c expects a C++ program.
